compiler-rt/lib/hwasan/hwasan_linux.cpp

```diff
@@ -106,8 +106,14 @@ static uptr GetHighMemEnd() {
 }
 
 static void InitializeShadowBaseAddress(uptr shadow_size_bytes) {
-  __hwasan_shadow_memory_dynamic_address =
-      FindDynamicShadowStart(shadow_size_bytes);
+  // NULL is generally address zero, so it is not a valid location for the
+  // shadow.
+  if (flags()->fixed_shadow_base != 0) {
+    __hwasan_shadow_memory_dynamic_address = flags()->fixed_shadow_base;
```

Comment:
oh,and maybe some test with use of the file?

Reply:
probably linux not android  only test

Reply:
Added compiler-rt/test/hwasan/TestCases/Linux/fixed-shadow.c

Reply:
For some reason, for static Android binaries, this branch gets taken